NextcloudCPack.cmake

```diff
@@ -2,11 +2,11 @@ include( InstallRequiredSystemLibraries )
 
 set( CPACK_PACKAGE_CONTACT  "Dominik Schmidt <domme@tomahawk-player.org>" )
 
-if ( DEFINED OEM_THEME_DIR AND EXISTS ${OEM_THEME_DIR}/OEM.cmake )
-     include ( "${OEM_THEME_DIR}/OEM.cmake" )
-else ()
-     include ( "${CMAKE_SOURCE_DIR}/NEXTCLOUD.cmake" )
-endif()
+#if ( DEFINED OEM_THEME_DIR AND EXISTS ${OEM_THEME_DIR}/OEM.cmake )
+#     include ( "${OEM_THEME_DIR}/OEM.cmake" )
+#else ()
+#     include ( "${CMAKE_SOURCE_DIR}/NEXTCLOUD.cmake" )
+#endif()
```

Comment:
Unrelated changes. We can remove that in a separate pr.

Reply:
It's not entirely unrelated. I meant to put this in its own commit so it could more easily be separated out. Mainly I wanted to draw attention to the fact that `NEXTCLOUD.cmake` or `OEM.cmake` gets run twice, which can cause complications with string transformations. If doing so doesn't break anything, commenting this out would allow moving the the suffix transformations out of `NEXTCLOUD.cmake` so that they could apply to `OEM.cmake` as well.

Reply:
I think it's fine to do the modifiying of the variables in `NEXTCLOUD.cmake`. Because it's related to this file. At a later point we could divide `NEXTCLOUD.cmake` in two sections. Like one where the user can make changes to the settings and the other for logic. After all I'm unsure why we even have this `NEXTCLOUD.cmake` file. For now I would just focus on getting the dev build done without to much changes.

Reply:
I definitely agree about further splitting the theme variables from any logic. FWIW, did you know that [CMake can actually process JSON](https://cmake.org/cmake/help/latest/command/string.html#json)? So idk hypothetically the theme variables could be served that way.

Reply:
I don't know if that is a great idea. I think the easiest way (for automatic builds) would be to allow to change these variables through environment variables.

Reply:
I think the idea with JSON is that it completely separates the data storage from the logic, which is a good way to avoid certain types of conflicts. That is, if the bare minimum set of name variables were in a JSON file, it would be all of a handful of lines, and it could be safely edited by a non-technical user. (If you're using the CMake GUI, things might work differently, but I don't have a strong understanding of how Nextcloud's automated builds work.)

Reply:
To be honest, I was mostly just pointing out that the capability exists.